lib/foreman_remote_execution_core/fake_script_runner.rb

```diff
@@ -0,0 +1,82 @@
+module ForemanRemoteExecutionCore
+  class ScriptRunner < ForemanTasksCore::Runner::Base
```

Comment:
Using the same name for two implementations might not be the best idea (if nothing else, it would be hard to test), what about putting the decision what runner to use into https://github.com/theforeman/foreman_remote_execution/blob/master/lib/foreman_remote_execution_core/actions.rb#L7 ?